### PR TITLE
Add scheme configuration for CNAME scheme

### PR DESCRIPTION
**Pull request by bleen**
See https://www.drupal.org/node/2514674

### CODE REVIEW

**Comment by bleen** [comment]
@deviantintegral it looks like the build is just timing out?  I'm pretty new to travis so maybe Im missing some other info about why this is failing... any pointers in terms of where I should look?

**Comment by deviantintegral** [comment]
Agreed, this looks like packagist or similar failed. I've restarted the build.

File: tests/StreamWrapperConfigurationTest.php [hunk 1]
```diff
@@ -60,6 +60,9 @@ public function testSetters() {
     $config->setDomain('cdn.example.com');
     $this->assertEquals('cdn.example.com', $config->getDomain());
 
+    $config->setDomainScheme('http');
```
Comment (deviantintegral) [inline tests/StreamWrapperConfigurationTest.php:64]: Can we also test that the default scheme is https?

File: amazons3.admin.inc [hunk 2]
```diff
@@ -59,6 +59,21 @@ function amazons3_admin() {
     '#default_value' => variable_get('amazons3_cname', 0),
   );
 
+  $schemes = array(
+    'https' => 'https',
+    'http' => 'http'
+  );
+  $form['amazons3_domain_scheme'] = array(
+    '#type' => 'select',
+    '#title' => t('CDN Domain Scheme'),
```
Comment (deviantintegral) [inline amazons3.admin.inc:68]: Let's add a #description to this field explaining what it's for.

Comment (deviantintegral) [inline amazons3.admin.inc:68]: And to clarify - what I mean is if I'm new to all of this, I wouldn't know if this refers to the backend S3 call or the URL rendered to clients.

**Comment by quicksketch** [comment]
The description has been added. This looks good to me. Thanks @bleen! Merged into 7.x-2.x.

**Pull request merged**